airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.kt

```diff
@@ -402,26 +406,76 @@ abstract class BaseSqlGeneratorIntegrationTest<DestinationState : MinimumDestina
      */
     @Test
     @Throws(Exception::class)
-    fun allTypes() {
-        // Add case-sensitive columnName to test json path querying
-        incrementalDedupStream.columns!!.put(
-            generator.buildColumnId("IamACaseSensitiveColumnName"),
-            AirbyteProtocolType.STRING
+    fun allTypes_safeCast() {
+        assumeTrue(
+            supportsSafeCast,
+            "Skipping test because this connector does not support safe casting"
         )
         createRawTable(streamId)
         createFinalTable(incrementalDedupStream, "")
         insertRawTableRecords(
             streamId,
-            BaseTypingDedupingTest.Companion.readRecords("sqlgenerator/alltypes_inputrecords.jsonl")
+            Streams.concat(
+                    BaseTypingDedupingTest.readRecords("sqlgenerator/alltypes_inputrecords.jsonl")
+                        .stream(),
+                    BaseTypingDedupingTest.readRecords(
+                            "sqlgenerator/safe_cast/alltypes_inputrecords.jsonl"
+                        )
+                        .stream()
+                )
+                .toList()
         )
 
-        var initialState =
-            getOnly(
-                destinationHandler.gatherInitialState(java.util.List.of(incrementalDedupStream))
-            )
-        Assertions.assertTrue(
-            initialState.isFinalTableEmpty,
-            "Final table should be empty before T+D"
+        executeTypeAndDedupe(
+            generator,
+            destinationHandler,
+            incrementalDedupStream,
+            Optional.empty(),
+            ""
+        )
+
+        verifyRecords(
+            "sqlgenerator/alltypes_expectedrecords_raw.jsonl",
+            dumpRawTableRecords(streamId),
+            "sqlgenerator/alltypes_expectedrecords_final.jsonl",
+            dumpFinalTableRecords(streamId, "")
+        )
+    }
+
+    /** Run a basic test to verify that we don't throw an exception on basic data values. */
+    @Test
+    @Throws(Exception::class)
+    fun allTypes_safeCast_handleGoodData() {
```

Comment:
is this uh... camel and snake case mixture a kotlin thing?

Reply:
it's a me thing :P I think the "Correct" way is to do
```kotlin
// Tests verifying correct handling of all data types
class AllTypes {
  // Tests that require the connector to support safe casting
  class SafeCast {
    @Test
    fun handleGoodData() { ...}
  }
}
```

but... that felt like I was just adding two levels of indentation for no benefit. Not a strong opinion though.